### PR TITLE
generation of basin codes

### PR DESCRIPTION
**Pull request by raphaeldussin**
starting a PR so you can try the updated code.
Things that I'd like to add:

- [x] updating the polygon for high resolution grids
- [x] adding a parser to main so we can use it as a standalone script 
- [x] docstrings,...

### CODE REVIEW

File: basins.py [hunk 1]
```diff
@@ -2,13 +2,32 @@
...
+ind_ATL = 2  # Atlantic Ocean
+ind_PAC = 3  # Pacific Ocean
+ind_ARC = 4  # Arctic Ocean
+ind_IND = 5  # Indian Ocean
+ind_MED = 6  # Mediterranean Sea
+ind_BLK = 7  # Black Sea
+ind_HUD = 8  # Hudson Bay
+ind_BAL = 9  # Baltic Sea
+ind_RED = 10 # Red Sea
+ind_PER = 11 # Persian Gulf
```
Comment (jkrasting) [inline basins.py]: The CMIP6/CF standard (Griffins et al. 2016) does not have a separate code for the Persian Gulf, although we’ve historically kept one in OM4 development.  We need an option to add the Persian Gulf points to the Indian basin if a user requests it.

File: basins.py [hunk 2]
```diff
@@ -183,22 +202,65 @@
...
+        codes.fillna(0)
+
+    codes['lon'] = lon_raw
+    return codes
+
+
+
+if __name__ == '__main__':
+
+    # define 1x1 grid
```
Comment (jkrasting) [inline basins.py:368]: @raphaeldussin - should we add a tripolar example to this?

**Review by jkrasting — CHANGES_REQUESTED** [review]

File: basins.py (hunk 2 above)
Comment (raphaeldussin) [reply basins.py:368]: I'd like to add a parser to the main and write tests for one degree (generated on the fly) and OM4 (that we can download
from the GFDL ftp)

**Comment by raphaeldussin** [comment]
@jkrasting polygons are ready! I'd suggest we merge as is and I'll make a second PR where I add a setup.py, argument parser and tests so that a more bite-size PR

File: basins.py [hunk 3]
```diff
@@ -203,36 +295,75 @@
...
+        ind_BLK,
+        ind_HUD,
+        ind_BAL,
+        ind_RED,
+        ind_PER,
+    ],
 )
 
 
+def generate_basin_codes(grid, lon="geolon", lat="geolat", mask="wet", persian=True):
```
Comment (jkrasting) [inline basins.py]:
```suggestion
def generate_basin_codes(grid, lon="geolon", lat="geolat", mask="wet", persian=False):
```

We'll thank ourselves the next time we run CMIP if we set this to False now.

File: basins.py [hunk 4]
```diff
@@ -203,36 +295,75 @@
...
+    lon : str
+        name of longitude array in grid
+
+    lat : str
+        name of latitude array in grid
+
+    mask : str
+        name of mask array in grid
+
+    persian : bool (default=True)
```
Comment (jkrasting) [inline basins.py]:
```suggestion
    persian : bool (default=False)
```

**Review by jkrasting — CHANGES_REQUESTED** [review]
Looks good @raphaeldussin.  Just one minor suggestion - LMK if you agree.

**Review by jkrasting — APPROVED** [review]

**Pull request merged**